_posts/2017-08-01-introducing-lumagl-v4.md

```diff
@@ -0,0 +1,202 @@
+
+
+
+### WebGL Capability Management
+
+* Dramatically simplifies building apps that run on both WebGL1 and WebGL2, seamlessly leveraging extensions when available.
+* Helps apps to query if a WebGL feature is available on the current platform - regardless of whether it is available through WebGL2 or through a WebGL extension.
+* When a feature can be provided either through WebGL2 or through a WebGL1 extension, luma.gl provides a single API that transparently uses the available implementation.
+
+
+### WebGL State Management
+
+* Enables apps to work with WebGL context state without having to worry about global side effects, addressing one of the major weak spots of the WebGL API.
+* Lets apps temporarily change global context state without having to do expensive queries to remember what values to restore it to.
+* Tracks changes to the context happening outside of luma.gl to ensure that global state always remains synchronized.
+* Prevents unnecessary calls to set state to current value.
+
+
+### GLSL Module System
+
+* The `ShaderAssembler` system allows shader code to be split into composable pieces.
+* It is completely optional, the application can use raw shader strings, `glslify`, the `ShaderAssembler` system or any other tools in isolation or combination to generate its shaders.
+* Optionally integrates with a `ShaderCache` to ensure textually equivalent shaders are only compiled once.
+
+
+### Seer Integration (Debug and Profiling)
+
+Debugging GLSL shaders can be one of the more frustrating part of WebGL programming, and from the start, luma.gl has included strong debugging support (including optionally logging values attributes and uniforms before draw calls, throwing exceptions showing the exact location of parsing errors in GLSL source code etc). In v4 we have taken this to the next level by integrating with the Seer chrome plugin. It is now possible to list and inspect all your `Model` instances directly in the Chrome dev tools, check attributes and uniforms, and even see performance timings from the last draw calls directly in the debugger.
+
+
+
+## WebGL2 API hightlights
+
+* WebGL2 constants added to `GL` export
+
+* Textures
+    * Can now created from `WebGLBuffers` in addition to typed arrays
+    * Tons of new texture formats
+    * Compressed textures from
+
+    * GLSL `dFdx`, `dFdy` Texture derivatives - (e.g. to compute normals in fragment shader)
+    * GLSL `texelFetch` - (e.g. for manual bilinear filtering)
+    * GLSL `textureGrad` - (e.g. for tweaking mipmap levels)
+    * Immutable texture?
+    * Integer texture - uint sampler
+    * Texture LOD
+    * GLSL `textureOffset`
+    * pixelStore
+    * srbg
+    * texture vertex (e.g. for displacement mapping)
+
+* Vertex Formats (GL.HALF_FLOAT)
+
+* GLSL
+    * centroid
+    * discard
+    * flat_smooth_interpolators
+    * non_square_matrix
+
+* TBD
+    * Uniform buffers
+
+* Misc
+    * New blending modes: `GL.MIN` and `GL.MAX`
+
+* Efficiency
```

Comment:
does _Efficiency_ here (under the _WebGL2 API highlight_) indicates better dev efficiency or runtime efficiency?